components/brave_wallet/browser/fil_nonce_tracker_unittest.cc

```diff
@@ -0,0 +1,154 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/fil_nonce_tracker.h"
+
+#include <utility>
+
+#include "base/strings/string_number_conversions.h"
+#include "base/test/bind.h"
+#include "base/test/task_environment.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_prefs.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_utils.h"
+#include "brave/components/brave_wallet/browser/fil_tx_meta.h"
+#include "brave/components/brave_wallet/browser/fil_tx_state_manager.h"
+#include "brave/components/brave_wallet/browser/json_rpc_service.h"
+#include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
+#include "components/sync_preferences/testing_pref_service_syncable.h"
+#include "services/data_decoder/public/cpp/test_support/in_process_data_decoder.h"
+#include "services/network/public/cpp/weak_wrapper_shared_url_loader_factory.h"
+#include "services/network/test/test_url_loader_factory.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_wallet {
+
+class FilNonceTrackerUnitTest : public testing::Test {
+ public:
+  FilNonceTrackerUnitTest()
+      : task_environment_(base::test::TaskEnvironment::TimeSource::MOCK_TIME) {
+    shared_url_loader_factory_ =
+        base::MakeRefCounted<network::WeakWrapperSharedURLLoaderFactory>(
+            &url_loader_factory_);
+  }
+
+  void SetUp() override {
+    brave_wallet::RegisterProfilePrefs(prefs_.registry());
+  }
+
+  PrefService* GetPrefs() { return &prefs_; }
+
+  network::SharedURLLoaderFactory* shared_url_loader_factory() {
+    return url_loader_factory_.GetSafeWeakWrapper().get();
+  }
+
+  void SetTransactionCount(uint64_t count) {
+    transaction_count_ = count;
+    url_loader_factory_.ClearResponses();
+
+    url_loader_factory_.AddResponse(
+        brave_wallet::GetNetworkURL(GetPrefs(), mojom::kLocalhostChainId,
+                                    mojom::CoinType::FIL)
+            .spec(),
+        GetResultString());
+  }
+  void WaitForResponse() { task_environment_.RunUntilIdle(); }
+
+  bool GetNextNonce(FilNonceTracker* tracker,
+                    const std::string& address,
+                    uint64_t expected_nonce) {
+    bool callback_called = false;
+    bool callback_result = false;
+    tracker->GetNextNonce(
+        address, base::BindLambdaForTesting([&](bool success, uint256_t nonce) {
+          callback_called = true;
+          callback_result = success;
+          EXPECT_EQ(expected_nonce, nonce);
+        }));
+    WaitForResponse();
+    EXPECT_TRUE(callback_called);
+    return callback_result;
+  }
```

Comment:
Same comment I left for eth_nonce_tracker_unittest.cc here.

"Instead of using callback_called and WaitForResponse, you can just use base::RunLoop here directly, and do run_loop.Quit(); inside lambda.
And no need to return callback_result, you can pass expected_success in and do EXPECT_EQ(expected_success, success) in lambda.` "

Reply:
done